### DISCUSSION TITLE
The right way to query mass of data(sqlite)

### DISCUSSION THREAD

**Discussion by ohperhaps** [sqlalchemy/sqlalchemy#6597] [Usage Questions] [answered]
Hi,
```
for user in User.query.filter_by(column_A='xxx', column_B='xxx'):
    for record in user.records:
        #do something
```

I need to loop records of users, 
but this loop takes long time to complete because of mass of users.
And my app is started by gunicorn with several worker processes.
So some of worker get Exception `Database is locked`.
can I do something make query completed(not lock the db) before the loop?
for example, If I add `all()` after the query like
`User.query.filter_by(column_A='xxx', column_B='xxx').all()`
Does this work?

Thanks

**Comment by CaselIT** [comment, marked as answer]
Hi,
If records is a relationship, you may want to look at the eager loading strategies in the docs https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html to avoid the n+1 problem

**Reply by ohperhaps** [reply]
Thank you for the relay.
The records is the relationship, and after I read docs,
I think I need `lazy='joined'`

**Reply by CaselIT** [reply]
Selectinload is usually preferred. But you may try both to see which one works best for you

**Reply by ohperhaps** [reply]
ok,I will both try joinedload and selectinload.Thank you again.
